app/MacPorts/config.py

```diff
@@ -0,0 +1,10 @@
+import os
+
+# configuration for rsync
+RSYNC = '/usr/bin/rsync'
+RSYNC_SOURCE = "rsync://rsync.macports.org/macports//trunk/dports/PortIndex_darwin_18_i386/PortIndex.json"
+JSON_FILE = os.path.join(os.path.abspath(os.path.dirname(__name__)), 'portindex.json')
+
+# configuration for repository to find updated ports
+REPO = "macports-ports"
+REPO_URL = "https://github.com/macports/" + REPO + ".git"
```

Comment:
I don't super agree with @umeshksingla about splitting the repository URL this way. One repository usually has one path, it can be `ssh` path to some other exotic place.

This splitting would make sense if you need to use 10 repositories from the same organisation throughout your app. Here I see little to no sense in doing this splitting. But more importantly, the name `REPO` is not too descriptive when put to a separate configuration file (it could be repo of webapp itself, for example). How about the following?
```
MACPORTS_PORTS_URL = "https://github.com/macports/macports-ports.git"
```

I later noticed that you use `REPO` in `os.path.isdir(config.REPO)` or `REPO_DIR = os.path.join(root, config.REPO)` (which is also a tiny bit inconsistent).

Just define some kind of `DATA_DIR` (feel free to use a better name than that) and something like
```
MACPORTS_PORTS_DIR = os.path.join(DATA_DIR, 'macports-ports')
```
and also use this in
```
git clone $MACPORTS_PORTS_URL $MACPORTS_PORTS_DIR
```
(with proper quoting, the above is just pseudocode).

Reply:
I have probably addressed this, thank you.